### DISCUSSION TITLE
Equivalent for `PHPStan\Testing\assertType` in Psalm?

### DISCUSSION THREAD

**Discussion by nikophil** [vimeo/psalm#11131] [Q&A] [answered]
Hello,

while developing external libraries, it is useful to see that static analyzers understand the code as expected.
For PHPStan, we're anlyzing a stub file which calls function `PHPStan\Testing\assertType`. ie: https://github.com/zenstruck/foundry/blob/2.x/stubs/functions.php

I'd like to do the same thing with Psalm, is there some equivalent tool?

thanks!

**Comment by weirdan** [comment, marked as answer]
There are several docblock tags that show / check types:

* https://psalm.dev/docs/annotating_code/supported_annotations/#psalm-trace - displays the inferred type (as a separate issue)
* https://psalm.dev/docs/annotating_code/supported_annotations/#psalm-check-type - performs a type assertion (`$bar = int` means the type of `$bar` is a subtype of `int`).
* https://psalm.dev/docs/annotating_code/supported_annotations/#psalm-check-type-exact - similar to the above, but requires an exact match. It will fail on `1 = int`, unlike `@psalm-check-type` that would pass.

**Reply by nikophil** [reply]
Hi @weirdan ,
Thanks for your answer, `@psalm-ckeck-type` is definitely what I was looking for!
